packages/sequelize/src/index.ts

```diff
@@ -0,0 +1,156 @@
+import type { Account as ApadterAccount } from "next-auth"
+import type {
+  Adapter,
+  AdapterUser,
+  AdapterSession,
+  VerificationToken,
+} from "next-auth/adapters"
+import { Sequelize, Model, ModelCtor } from "sequelize"
+import {
+  accountSchema,
+  userSchema,
+  sessionSchema,
+  verificationTokenSchema,
+} from "./schema"
+
+// @see https://sequelize.org/master/manual/typescript.html
+interface AccountInstance
+  extends Model<ApadterAccount, Partial<ApadterAccount>>,
+    ApadterAccount {}
+interface UserInstance
+  extends Model<AdapterUser, Partial<AdapterUser>>,
+    AdapterUser {}
+interface SessionInstance
+  extends Model<AdapterSession, Partial<AdapterSession>>,
+    AdapterSession {}
+interface VerificationTokenInstance
+  extends Model<VerificationToken, Partial<VerificationToken>>,
+    VerificationToken {}
+
+export { accountSchema, userSchema, sessionSchema, verificationTokenSchema }
+
+export default function SequelizeAdapter(client: Sequelize): Adapter {
+  const tableOptions = { underscored: true, timestamps: false }
+
+  const Account =
+    (client.models.Account as ModelCtor<any>) ||
+    client.define<AccountInstance>("account", accountSchema, tableOptions)
+  const User =
+    (client.models.User as ModelCtor<any>) ||
+    client.define<UserInstance>("user", userSchema, tableOptions)
+  const Session =
+    (client.models.Session as ModelCtor<any>) ||
+    client.define<SessionInstance>("session", sessionSchema, tableOptions)
+  const VerificationToken =
+    (client.models.VerificationToken as ModelCtor<any>) ||
+    client.define<VerificationTokenInstance>(
+      "verificationToken",
+      verificationTokenSchema,
+      tableOptions
+    )
```

Comment:
```suggestion
import type { Sequelize, Model, ModelCtor, ModelOptions } from "sequelize"
import * as defaultModels from "./schema"

export { defaultModels as models }

// @see https://sequelize.org/master/manual/typescript.html
interface AccountInstance
  extends Model<ApadterAccount, Partial<ApadterAccount>>,
    ApadterAccount {}
interface UserInstance
  extends Model<AdapterUser, Partial<AdapterUser>>,
    AdapterUser {}
interface SessionInstance
  extends Model<AdapterSession, Partial<AdapterSession>>,
    AdapterSession {}
interface VerificationTokenInstance
  extends Model<VerificationToken, Partial<VerificationToken>>,
    VerificationToken {}

interface SequelizeAdapterOptions {
  models?: Partial<{
    User: ModelCtor<UserInstance>
    Account: ModelCtor<AccountInstance>
    Session: ModelCtor<SessionInstance>
    VerificationToken: ModelCtor<VerificationTokenInstance>
  }>
}

export default function SequelizeAdapter(
  client: Sequelize,
  options?: SequelizeAdapterOptions
): Adapter {
  const { models } = options ?? {}

  const defaultModelOptions: ModelOptions = {
    underscored: true,
    timestamps: false,
  }

  const { User, Account, Session, VerificationToken } = {
    User:
      models?.User ??
      client.define<UserInstance>(
        "user",
        defaultModels.User,
        defaultModelOptions
      ),
    Account:
      models?.Account ??
      client.define<AccountInstance>(
        "account",
        defaultModels.Account,
        defaultModelOptions
      ),
    Session:
      models?.Session ??
      client.define<SessionInstance>(
        "session",
        defaultModels.Session,
        defaultModelOptions
      ),
    VerificationToken:
      models?.VerificationToken ??
      client.define<VerificationTokenInstance>(
        "verificationToken",
        defaultModels.VerificationToken,
        defaultModelOptions
      ),
  }
```

I would refactor this section in a way that makes it easier to configure by the user.

With my suggestions, the user would override `User` like this:

```js
SequelizeAdapter(client, {
  schemas: {
    User: client.define("user", MyUserModel)
  }
})
```

What do you think?

Also for consistent naming, I think we should just stick to models/model instead of schemas, as the TS types suggest that anyway. (could you rename the `schema.ts` file to `models.ts` as well?)

Reply:
Yep I think that's a fair change, I've refactored that in f6b9518 I've also added docs on how to do this in cd44414